### PR TITLE
fix(app): naïve loading state for tc open lid during lw cal

### PR DESCRIPTION
**Pull request by b-cooper**
## overview

In the absence of live status updates while the lid is opening or closing (limitation of the TC firmware), this adds a naïve loading state that delays a set amount of time in order to show to the user that the task has begun.

## review requests

Enter Labware Calibration for a protocol that involves a TC, and make sure the lid of the thermocycler is closed.  When prompted to "Open Lid" the loading state should appear (disabled button and spinner)

### CODE REVIEW

File: app/src/components/PrepareModules/index.js [hunk 1]
```diff
@@ -28,12 +31,10 @@ function PrepareModules(props: Props) {
             sendModuleCommand(robot, mod.serial, { command_type: 'open' })
           )
       )
+    setIsHandling(true)
+    setTimeout(() => setIsHandling(false), LID_OPEN_DELAY_MS)
   }
 
-  const isHandling = some(
```
Comment (mcous) [inline app/src/components/PrepareModules/index.js]: We might want to keep this around (commented out) with a TODO so we don't lose context here

File: app/src/components/PrepareModules/index.js [hunk 2]
```diff
@@ -28,12 +31,10 @@ function PrepareModules(props: Props) {
             sendModuleCommand(robot, mod.serial, { command_type: 'open' })
           )
       )
+    setIsHandling(true)
+    setTimeout(() => setIsHandling(false), LID_OPEN_DELAY_MS)
```
Comment (mcous) [inline app/src/components/PrepareModules/index.js]: We need cleanup on this timeout or else will have issues if the user navigates away during the timeout, the component unmounts, and then the timeout attempts to call `setIsHandling`

Adapting our current `useInterval` hook into a `useTimeout` hook (with tests) might be worth the time:

```js
// useTimeout.js
// @flow
import { useEffect, useRef } from 'react'

/**
 * React hook to call a function on a timeout; adapted from:
 * https://overreacted.io/making-setinterval-declarative-with-react-hooks/
 *
 * @template T (type of the input value)
 * @param {() => mixed} callback (function to call on an interval)
 * @param {number | null} delay (timeout delay, or null to disable the timeout)
 * @returns {void}
 */
export function useTimeout(callback: () => mixed, delay: number | null): void {
  const savedCallback = useRef()

  // remember the latest callback
  useEffect(() => {
    savedCallback.current = callback
  }, [callback])

  // set up the interval
  useEffect(() => {
    const tick = () => savedCallback.current && savedCallback.current()
    if (delay !== null) {
      const id = setTimeout(tick, delay)
      return () => clearTimeout(id)
    }
  }, [delay])
}
```

Then:

```js
const handleOpenLidClick = () => {
  // ...
  setIsHandling(true)
}

useTimeout(
  () => setIsHandling(false),
  isHandling ? LID_OPEN_DELAY_MS : null
)
```

File: components/src/hooks/useTimeout.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+// @flow
+import { useEffect, useRef } from 'react'
+
+/**
+ * React hook to call a function on an interval; copied from:
```
Comment (mcous) [inline components/src/hooks/useTimeout.js]: Slight copy-paste typo

File: components/src/hooks/__tests__/useInterval.test.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ import * as React from 'react'
 import { mount } from 'enzyme'
 import { useInterval } from '..'
 
-describe('usePrevious hook', () => {
+describe('useInterval hook', () => {
```
Comment (mcous) [inline components/src/hooks/__tests__/useInterval.test.js]: Doh

File: app/src/components/CalibrateLabware/ProceedToRun.js [hunk 5]
```diff
@@ -32,6 +33,7 @@ function InfoBoxButton(props: Props) {
   const handleClick = () => {
     // $FlowFixMe: robotActions.returnTip is not typed
     returnTip()
+    moveGantryToSafety()
```
Comment (mcous) [inline app/src/components/CalibrateLabware/ProceedToRun.js]: Are these RPC actions safe to dispatch one after another like this? The RPC client isn't smart enough to queue these or anything

Comment (b-cooper) [reply app/src/components/CalibrateLabware/ProceedToRun.js]: these changes snuck in from another working branch, where I'm trying to fix up the place pcr seal prompt problems. I'll pull them out of this one. Good catch

**Pull request merged**